src/wp-includes/class-wp-duotone.php

```diff
@@ -0,0 +1,1064 @@
+				self::$block_css_declarations,
+				array(
+					'context' => 'block-supports',
+				)
+			);
+		}
+	}
+
+	/**
+	 * Appends the used global style duotone filter presets (CSS custom
+	 * properties) to the inline global styles CSS.
+	 *
+	 * @since 6.3.0
+	 */
+	public static function output_global_styles() {
+		if ( ! empty( self::$used_global_styles_presets ) ) {
+			wp_add_inline_style( 'global-styles', self::get_global_styles_presets( self::$used_global_styles_presets ) );
+		}
+	}
+
+	/**
+	 * Outputs all necessary SVG for duotone filters, CSS for classic themes.
+	 *
+	 * @since 6.3.0
+	 */
+	public static function output_footer_assets() {
+		if ( ! empty( self::$used_svg_filter_data ) ) {
+			echo self::get_svg_definitions( self::$used_svg_filter_data );
+		}
+
+		// This is for classic themes - in block themes, the CSS is added in the head via wp_add_inline_style in the wp_enqueue_scripts action.
+		if ( ! wp_is_block_theme() && ! empty( self::$used_global_styles_presets ) ) {
+			wp_add_inline_style( 'core-block-supports', self::get_global_styles_presets( self::$used_global_styles_presets ) );
+		}
+	}
+
+	/**
+	 * Adds the duotone SVGs and CSS custom properties to the editor settings so
+	 * they can be pulled in by the EditorStyles component in JS and rendered in
+	 * the post editor.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param array $settings The block editor settings from the `block_editor_settings_all` filter.
+	 * @return array The editor settings with duotone SVGs and CSS custom properties.
+	 */
+	public static function add_editor_settings( $settings ) {
+		if ( ! empty( self::$global_styles_presets ) ) {
+			if ( ! isset( $settings['styles'] ) ) {
+				$settings['styles'] = array();
+			}
+
+			$settings['styles'][] = array(
+				// For the editor we can add all of the presets by default.
+				'assets'         => self::get_svg_definitions( self::$global_styles_presets ),
+				// The 'svgs' type is new in 6.3 and requires the corresponding JS changes in the EditorStyles component to work.
+				'__unstableType' => 'svgs',
+				// These styles not generated by global styles, so this must be false or they will be stripped out in wp_get_block_editor_settings.
+				'isGlobalStyles' => false,
+			);
+
+			$settings['styles'][] = array(
+				// For the editor we can add all of the presets by default.
+				'css'            => self::get_global_styles_presets( self::$global_styles_presets ),
+				// This must be set and must be something other than 'theme' or they will be stripped out in the post editor <Editor> component.
+				'__unstableType' => 'presets',
+				// These styles are no longer generated by global styles, so this must be false or they will be stripped out in wp_get_block_editor_settings.
+				'isGlobalStyles' => false,
+			);
+		}
+
+		return $settings;
+	}
+
+	/**
+	 * Migrate the old experimental duotone support flag to its stabilized location
+	 * under `supports.filter.duotone` and sets.
```

Comment:
Again this needs to be split in to a summary and longer description/

When the developer docs are generated the first line is a short summary displayed below the title, followed by a longer description that goes in to detail. See https://developer.wordpress.org/reference/classes/wp_query/get_posts/ fpr example.

:1234: This chnage is needed in numerous places so I'll get you to reeveluate them rather than continue adding noise to the PR

Reply:
Updated in [0b06740](https://github.com/WordPress/wordpress-develop/pull/4619/commits/0b067400ce2dd3c95cc955315469ff973cb61b4a).